gtsfm/utils/io.py

```diff
@@ -291,14 +347,28 @@ def write_images(gtsfm_data: GtsfmData, images: List[Image], save_dir: str) -> N
 
     num_imgs = gtsfm_data.number_images()
     # TODO: compute this (from keypoint data? or from track data?)
-    mean_obs_per_img = 0
+
+    image_id_num_measurements = {}
+    for j in range(gtsfm_data.number_tracks()):
+        track = gtsfm_data.get_track(j)
+        for k in range(track.number_measurements()):
+            image_id, uv_measured = track.measurement(k)
+            if image_id not in image_id_num_measurements:
+                image_id_num_measurements[image_id] = 1
+            else:
+                image_id_num_measurements[image_id] += 1
```

Comment:
you can avoid this if-else by using a defaultdict

Reply:
Now a defaultdict